src/python/pants/option/options_fingerprinter.py

```diff
@@ -5,16 +5,26 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
+import inspect
 import json
 import os
 from hashlib import sha1
 
 from pants.base.build_environment import get_buildroot
-from pants.option.custom_types import dict_with_files_option, file_option, target_option
+from pants.option.custom_types import UnsetBool, dict_with_files_option, file_option, target_option
+
+
+class Encoder(json.JSONEncoder):
+  _UNSET_BOOL_ENCODING = '__type::{}::{}'.format(inspect.getmodule(UnsetBool), UnsetBool.__name__)
```

Comment:
> Well, as long as it's not a performance liability ....
Its a class constant, so no real concern
> Do we actually care what module this is in though?
No, just that it doesn't collide with other strings. Open to a concrete final suggestion on what value makes sense.